tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1303 @@
+      check:
+        not node.isSubscribed(pubsubTopic)
+        node.subscribedTopics != pubsubTopicSeq
+        otherNode.isSubscribed(pubsubTopic)
+        otherNode.subscribedTopics == pubsubTopicSeq
+
+      await sleepAsync(500.millis)
+
+      # When publishing a message in the subscribed node
+      let fromOtherWakuMessage = fakeWakuMessage("fromOther")
+      discard await otherNode.publish(pubsubTopic, fromOtherWakuMessage)
+
+      # Then the message is published only in the subscribed node
+      check:
+        not await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+
+      let (otherTopic1, otherMessage1) = otherHandlerFuture.read()
+      check:
+        otherTopic1 == pubsubTopic
+        otherMessage1 == fromOtherWakuMessage
+      
+      # When publishing a message in the other node
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      let fromNodeWakuMessage = fakeWakuMessage("fromNode")
+      discard await node.publish(pubsubTopic, fromNodeWakuMessage)
+
+      # Then the message is published only in the subscribed node
+      check:
+        not await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+      
+      let (otherTopic2, otherMessage2) = otherHandlerFuture.read()
+      check:
+        otherTopic2 == pubsubTopic
+        otherMessage2 == fromNodeWakuMessage
+      
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+    asyncTest "Pubsub Topic Subscription (Network Size: 2, both subscribed to same pubsub topic)":
+      # Given a second node connected to the first one
+      let 
+        otherSwitch = newTestSwitch()
+        otherNode = await newTestWakuRelay(otherSwitch)
+
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let otherRemotePeerInfo = otherSwitch.peerInfo.toRemotePeerInfo()
+      check await peerManager.connectRelay(otherRemotePeerInfo)
+
+      var otherHandlerFuture = newPushHandlerFuture()
+      proc otherSimpleFutureHandler(topic: PubsubTopic, message: WakuMessage) {.async, gcsafe.} =
+        otherHandlerFuture.complete((topic, message))
+
+      # When subscribing both nodes to the same Pubsub Topic
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+
+      # Then both nodes are subscribed
+      check:
+        node.isSubscribed(pubsubTopic)
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.isSubscribed(pubsubTopic)
+        otherNode.subscribedTopics == pubsubTopicSeq
+
+      await sleepAsync(500.millis)
+
+      # When publishing a message in node
+      let fromOtherWakuMessage = fakeWakuMessage("fromOther")
+      discard await node.publish(pubsubTopic, fromOtherWakuMessage)
+
+      # Then the message is published only in both nodes
```

Comment:
I think the comment has a copy&paste error:)

Reply:
Indeed 😅 You repeat one thing too many times and you start missing obvious errors